lib/auth/in_memory/indexer.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Class that provides an internal indexing over the simple data provided by
+ * the authentication configuration file for the memory backend. This allows
+ * accessing the different authentication entities through various types of
+ * keys.
+ *
+ * @class Indexer
+ */
+module.exports = class Indexer {
+    /**
+     * @constructor
+     * @param {object} authdata - the authentication config file's data
+     * @return {undefined}
+     */
+    constructor(authdata) {
+        this.accountsBy = {
+            canId: {},
+            accessKey: {},
+            email: {},
+        };
+        this.usersBy = {
+            accessKey: {},
+            email: {},
+        };
+
+        /*
+         * This may happen if the application is configured to use another
+         * authentication backend than in-memory.
+         * As such, we're managing the error here to avoid screwing up there.
+         */
+        if (!authdata) {
```

Comment:
I think we always start up the in-memory backend, and initialize the indexer, even if the service is configured to use actual Vault and there is no `config.authData`. So we don't want to crash the service in that case.